Compile error dereferencing a pct_string_view without including decode_view.hpp

Dereferencing a pct_string_view when decode_view.hpp is not included:

```
...\router.cpp(91,12): error C2027: use of undefined type 'boost::urls::decode_view'
      ...\boost\libs\url\include\boost\url\pct_string_view.hpp(31,7):
```

--- a/include/boost/url/pct_string_view.hpp
+++ b/include/boost/url/pct_string_view.hpp
@@ -464,4 +464,7 @@ to_sv(pct_string_view const& s) noexcept
 } // urls
 } // boost
 
+// Ensure decode_view is complete for operator*()
+#include <boost/url/decode_view.hpp>
+
 #endif

--- a/test/unit/pct_string_view.cpp
+++ b/test/unit/pct_string_view.cpp
@@ -134,10 +134,36 @@ struct pct_string_view_test
     }
 
     void
+    testOperatorStar()
+    {
+        // Basic usage
+        {
+            pct_string_view psv("Program%20Files");
+            decode_view dv = *psv;
+            BOOST_TEST(dv == "Program Files");
+        }
+
+        // Empty string
+        {
+            pct_string_view psv("");
+            decode_view dv = *psv;
+            BOOST_TEST(dv.empty());
+        }
+
+        // No encoding
+        {
+            pct_string_view psv("simple");
+            decode_view dv = *psv;
+            BOOST_TEST(dv == "simple");
+        }
+    }
+
+    void
     run()
     {
         testSpecial();
         testRelation();
+        testOperatorStar();
     }
 };
 

